link/syscalls.go

```diff
@@ -10,6 +10,22 @@ import (
 	"github.com/cilium/ebpf/internal/unix"
 )
 
+// Type is the kind of link.
+type Type uint32
+
+// Valid link types.
+//
+// Equivalent to enum bpf_link_type.
+const (
+	UnspecifiedType Type = iota
+	RawTracepointType
+	TracingType
+	CgroupType
+	IterType
+	NetNsType
+	XDPType
```

Comment:
Is that name unique enough? Why not `…LinkType`?

Reply:
Because it would read as `link.XDPLinkType` which seems a bit redundant.

Reply:
Makes sense. Didn't notice it's in its own package.